pkg/controller/stack/stack_controller.go

```diff
@@ -171,6 +170,15 @@ func (r *ReconcileStack) Reconcile(request reconcile.Request) (reconcile.Result,
 		return reconcile.Result{}, err
 	}
 
+	// Delete the working directory after the reconciliation is completed (regardless of success or failure).
+	defer func() {
+		if sess.workdir != "" {
+			if err := os.RemoveAll(sess.workdir); err != nil {
+				sess.logger.Error(err, "Failed to delete working dir: %s", sess.workdir)
+			}
+		}
+	}()
```

Comment:
Minor - I wonder if we should move this into a `sess.ClosePulumiWorkdir` method and put that near `SetupPulumiWorkdir` to make the creation/destruction more cleanly coupled?

Reply:
Agreed. Updated.